core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java

```diff
@@ -881,6 +880,7 @@ private void cleanUpTask(TaskState taskState)
             whenSplitQueueHasSpace.complete(null, executor);
         }
 
+        dynamicFiltersFetcher.stop();
```

Comment:
May be we do this in fatalUnacknowledgedFailure and failLocallyImmediately ?
In "normal" cases of failure/cancel/done I think we wouldn't need to do this.